src/draw_buffer.rs

```diff
@@ -35,13 +37,11 @@ impl ColorAttachment {
             ColorAttachment::Texture(_) => ColorAttachmentType::Texture,
         }
     }
-}
 
-impl Drop for ColorAttachment {
-    fn drop(&mut self) {
-        match *self {
-            ColorAttachment::Renderbuffer(id) => gl::delete_renderbuffers(&[id]),
-            ColorAttachment::Texture(tex_id) => gl::delete_textures(&[tex_id]),
+    fn destroy(self, gl: &gl::Gl) {
```

Comment:
Ah, you are passing `gl` in, I see. And the function is not public, so there is no concern about calling it twice.
Let's leave it with `destroy` then :)